packages/itwin/ec3-widget/src/test/test-utils.tsx

```diff
@@ -0,0 +1,86 @@
+import type { RenderResult } from "@testing-library/react";
+import { act } from "@testing-library/react";
+import { fireEvent, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import React from "react";
+import type { GetAccessTokenFn } from "../components/api/APIContext";
+import { ApiContext } from "../components/api/APIContext";
+import * as moq from "typemoq";
+
+export interface RenderParameters {
+  component: React.ReactNode;
+  ec3ConfigurationsClient?: IEC3ConfigurationsClient;
+  ec3JobsClient?: IEC3JobsClient;
+  reportsClient?: IReportsClient;
+  oDataClient?: IOdataClient;
+  getAccessTokenFn?: GetAccessTokenFn;
+}
+
+export async function renderWithContext({
+  component, ec3ConfigurationsClient, ec3JobsClient, reportsClient, oDataClient, getAccessTokenFn,
+}: RenderParameters): Promise<RenderResult> {
+  let result: RenderResult;
+  await act(async () => {
+    result = render(
+      <ApiContext.Provider value={{
+        reportsClient: reportsClient ?? moq.Mock.ofType<IReportsClient>().object,
+        oDataClient: oDataClient ?? moq.Mock.ofType<IOdataClient>().object,
+        ec3JobsClient: ec3JobsClient ?? moq.Mock.ofType<IEC3JobsClient>().object,
+        ec3ConfigurationsClient: ec3ConfigurationsClient ?? moq.Mock.ofType<IEC3ConfigurationsClient>().object,
+        getAccessTokenFn: getAccessTokenFn ?? (async () => ""),
+      }}>
+        {component}
+      </ApiContext.Provider>
+    );
+  });
+  return result!;
+}
+
+export async function simulateInput(rootElement: HTMLElement, text: string) {
+  const input = rootElement.querySelector(".iui-select-button") as HTMLInputElement;
+  await userEvent.click(input);
+  const item = screen.getByText(text);
+  await userEvent.click(item);
+  expect(input.querySelector(".iui-content")).toHaveTextContent(text);
+}
+
+export async function simulateCombobox(rootElement: HTMLElement, text: string) {
+  const input = rootElement.querySelector(".iui-input") as HTMLInputElement;
+  fireEvent.focus(input);
+  const item = screen.getByText(text);
+  await act(async () => {
+    await userEvent.click(item);
+  });
+  expect(input.value).toEqual(text);
+}
+
+export async function simulateTextInput(rootElement: HTMLElement, text: string) {
+  const input = rootElement as HTMLInputElement;
+  await act(async () => {
+    fireEvent.change(input, { target: { value: text } });
+  });
+  expect(input.value).toEqual(text);
+}
+
+export function getComboboxOptions(rootElement: HTMLElement) {
+  const input = rootElement.querySelector(".iui-input") as HTMLInputElement;
+  fireEvent.focus(input);
+  return document.querySelectorAll(".iui-menu-item");
+}
+
+export async function getInputOptions(rootElement: HTMLElement) {
+  const input = rootElement.querySelector(".iui-select-button") as HTMLInputElement;
+  await userEvent.click(input);
```

Comment:
You may want to wrap this `click` with `act` since it's a user input.

Reply:
Just to be on the safe side, I would wrap all events (click, focus, etc) with `act`.

Reply:
wrapped all ```.click``` ```.focus``` and ```.change``` functions